crates/miden-block-prover/src/local_block_prover.rs

```diff
@@ -0,0 +1,265 @@
+            block_note_tree,
+            created_nullifiers,
+            chain_mmr,
+            prev_block_header,
+        ) = proposed_block.into_parts();
+
+        let prev_block_commitment = prev_block_header.hash();
+
+        // Get the root of the block note tree.
+        // --------------------------------------------------------------------------------------------
+
+        // TODO: Do we need the full tree in proposed block or would the root be sufficient? We
+        // can reconstruct the tree from the output note batches, so the question is whether it
+        // should be included in the proven block to be readily accessible or if recomputing it is
+        // fine.
+        let note_root = block_note_tree.root();
+
+        // Insert the created nullifiers into the nullifier tree to compute its new root.
+        // --------------------------------------------------------------------------------------------
+
+        let (created_nullifiers, new_nullifier_root) =
+            compute_nullifiers(created_nullifiers, &prev_block_header, block_num)?;
+
+        // Insert the previous block header into the block chain MMR to get the new chain root.
+        // --------------------------------------------------------------------------------------------
+
+        let new_chain_root = compute_chain_root(chain_mmr, prev_block_header);
+
+        // Insert the state commitments of updated accounts into the account tree to compute its new
+        // root.
+        // --------------------------------------------------------------------------------------------
+
+        let new_account_root = compute_account_root(&mut account_updated_witnesses)?;
+
+        // Transform the account update witnesses into block account updates.
+        // --------------------------------------------------------------------------------------------
+
+        let updated_accounts = account_updated_witnesses
+            .into_iter()
+            .map(|(account_id, update_witness)| {
+                let (
+                    _initial_state_commitment,
+                    final_state_commitment,
+                    // Note that compute_account_root took out this value so it should not be used.
+                    _initial_state_proof,
+                    details,
+                    transactions,
+                ) = update_witness.into_parts();
+                BlockAccountUpdate::new(account_id, final_state_commitment, details, transactions)
+            })
+            .collect();
+
+        // Construct the new block header.
+        // --------------------------------------------------------------------------------------------
+
+        // TODO: Where is this defined? Should we rename this to `protocol_version`, if it is that?
+        let version = 0;
+        // TODO: How should we compute this? Which kernel do we mean (tx, batch, block)? Should we
+        // rename it to indicate that?
+        let kernel_root = Digest::default();
```

Comment:
This should come from `miden-lib`, specifically, `TransactionKernel::kernel_root()`. This is a commitment to the kernels which transactions in this block are allowed to use. The idea is that in the future we may (and probably will) make changes to the transaction kernel and as we do, these would be added to the kernel root.

For now, we have only 1 kernel version.